src/mercury_engine_data_structures/formats/bmsstoc.py

```diff
@@ -0,0 +1,69 @@
+import construct
+from construct.core import (
+    Array,
+    Byte,
+    Check,
+    Const,
+    Construct,
+    Int24ul,
+    Int32ul,
+    Int64ul,
+    PaddedString,
+    PrefixedArray,
+    Rebuild,
+    Struct,
+)
+
+from mercury_engine_data_structures.formats.base_resource import BaseResource
+from mercury_engine_data_structures.formats.property_enum import PropertyEnum
+from mercury_engine_data_structures.game_check import Game
+
+FileId = Struct(
+    "file_idx" / Int24ul,
+    "file_type" / construct.Enum(Byte, sound=1, soundgroup=2, bank=3, player=4, wavearchive=5, group=6),
+)
+
+WaveArchiveData = Struct(
+    Const(0x20, Int32ul),
+    "wav_name" / PropertyEnum,
+    "file_id" / FileId,
+    "warc_index" / Int32ul, # index of the WARC listed in BFSAR
+    "idx" / Int32ul, # 0 to max wave archives
```

Comment:
I don't understand this comment

Reply:
not quite sure the best way to write it but in the BFSAR all sounds have a "global" index which is `warc_index` (like, sounds are numbered `0 - (N-1)` and wave archives are `N - N+len(WArc)`. and the idx field is the same but starts from 0.